Sources/ETCDExample/ETCDExample.swift

```diff
@@ -13,32 +13,57 @@
 //===----------------------------------------------------------------------===//
 import ETCD
 import NIO
+import Dispatch
 
 @main
 struct Example {
     static func main() async throws {
+        let eventLoopGroup = MultiThreadedEventLoopGroup(numberOfThreads: 1)
         do {
-            let etcdClient = EtcdClient(host: "localhost", port: 2379, eventLoopGroup: MultiThreadedEventLoopGroup(numberOfThreads: 1))
+            let etcdClient = EtcdClient(host: "localhost", port: 2379, eventLoopGroup: eventLoopGroup)
+            Task {
```

Comment:
Instead of spawning an unstructured task I would recommend the following setup

```
try await withThrowingTaskGroup(of: Void.self) { group in
    group.addTask {
                do {
                    try await etcdClient.watch("foo") { sequence in
                        var iterator = sequence.makeAsyncIterator()
                        while let event = try await iterator.next() {
                            print(event)
                        }
                    }
                } catch {
                    print("Error watching key: \(error)")
                }
    }
    // Sleeping for a second the let the watch above setup
    try await Task.sleep(for: .seconds(1))

     // Your calls to set and get
}
```